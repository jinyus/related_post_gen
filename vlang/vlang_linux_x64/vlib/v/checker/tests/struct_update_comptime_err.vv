struct Aa {
	sub Bb
}

struct AaWithAliasType {
	sub Bb
}

type AliasType = Bb

struct Bb {
	a int
}

fn encode_struct[U](val U) {
	$for field in U.fields {
		_ = struct {
			...val.$(field.name)
		}
	}
}

fn main() {
	aa := Aa{}
	bb := Bb{}
	aa_with_alias_type := AaWithAliasType{}

	encode_struct(aa)
	encode_struct(aa_with_alias_type)

	encode_struct(bb)
}
