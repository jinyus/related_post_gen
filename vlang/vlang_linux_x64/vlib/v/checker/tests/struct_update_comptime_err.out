vlib/v/checker/tests/struct_update_comptime_err.vv:18:4: error: cannot use struct update syntax in compile time expressions
   16 |     $for field in U.fields {
   17 |         _ = struct {
   18 |             ...val.$(field.name)
      |             ~~~
   19 |         }
   20 |     }
